.github/workflows/ci.ml

```diff
@@ -0,0 +1,65 @@
+module List = ListLabels
+
+let skip_test =
+  match Sys.getenv "SKIP_TEST" with
+  | exception Not_found -> false
+  | s -> bool_of_string s
+
+let run cmd args =
+  let cmd = String.concat " " (cmd :: args) in
+  match Sys.command cmd with
+  | 0 -> ()
+  | n ->
+    Printf.eprintf "'%s' failed with code %d" cmd n;
+    exit n
+
+let opam args = run "opam" args
+
+let pin () =
+  let packages =
+    let packages = Sys.readdir "." |> Array.to_list in
+    let packages =
+      List.fold_left packages ~init:[] ~f:(fun acc fname ->
+          if Filename.check_suffix fname ".opam" then
+            let dot = String.rindex fname '.' in
+            String.sub fname 0 dot :: acc
+          else
+            acc)
+    in
+    if skip_test then
+      List.filter packages ~f:(fun pkg -> pkg = "dune")
+    else
+      packages
+  in
+  List.iter packages ~f:(fun package ->
+      opam [ "pin"; "add"; package ^ ".next"; "."; "--no-action" ])
+
+let test () =
+  if skip_test then
+    ()
+  else
+    let win = Sys.cygwin || Sys.win32 in
```

Comment:
The presence of cygwin is surprising - why can't Cygwin execute the other branch, or does it simply not work with the GitHub action at present?

Reply:
Well I just thought that test suite doesn't work on cygwin. Let's give it a try though.

Reply:
It's fine if it doesn't (because it's new) but that should be a TODO comment, because it _should_ work!